pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -182,6 +179,16 @@ type Status struct {
 	err     error
 }
 
+func NewStatus(reasons ...string) *Status {
```

Comment:
can this just be a private function in scheduler_test.go?

Reply:
and what was wrong with old way?

Reply:
The issue is that the test in scheduler_test used to construct just Status{} to indicate this is NoFit. However, now I equate nil Status and empty. So, in the scheduler_test I need to pass reasons, but reasons are not an exported field. Thus I create the construct which allows me not to export the reasons field.

I think ideally `Status`  as a type is not even exported. The only use of the type outside of flavorassigner is in this scheduler test. However, this seems out of scope for the PR.